### ISSUE TITLE
Incorrect work query of some EMBEDDED values

### ISSUE THREAD

**Issue by WeaxMe** [orientechnologies/orientdb#7795]
### OS: Ubuntu 16.04
Hi, I need to create filter by some values in document EMBEDDED field, but something is wrong.

## Steps to reproduce  
1. `CREATE CLASS TestEmbedded`
2. `CREATE PROPERTY TestEmbedded.name STRING`
3. `CREATE PROPERTY TestEmbedded.embedded EMBEDDED`
4. `INSERT INTO TestEmbedded SET name = "test_1", embedded = {"name": "test_1"}`
5. `INSERT INTO TestEmbedded SET name = "test_2", embedded = {"name": "test_2"}`
6. `SELECT FROM TestEmbedded` - returns two documents
7. `SELECT FROM TestEmbedded WHERE embedded IN (SELECT FROM (SELECT EXPAND(embedded) FROM $current) WHERE ANY() LIKE "test_1")` - returns document with name `"test_1"` - works fine.
8. `SELECT FROM TestEmbedded WHERE embedded IN (SELECT FROM (SELECT EXPAND(embedded) FROM $current) WHERE ANY() LIKE "test_2")` **- don't return something, but must return document with name `"test_2"` and embedded record `{"name": "test_2"}`**

If execute this queries all works fine:
1. `SELECT FROM (SELECT EXPAND(embedded) FROM TestEmbedded) WHERE ANY() LIKE "test_1"` - returns embedded record {"name":"test_1"}
2. `SELECT FROM (SELECT EXPAND(embedded) FROM TestEmbedded) WHERE ANY() LIKE "test_2"` - returns embedded record {"name":"test_2"}
3. `SELECT FROM (SELECT EXPAND(embedded) FROM TestEmbedded)` - returns two embedded records

Thanks, 
Vitaliy.

**Comment by WeaxMe** [comment]
Sorry, I write incorrect SQL query. 
Can you tell me how to write SQL query for getting result with documents which contains embedded field with given value?
Thanks,
Vitaliy.

**Comment by luigidellaquila** [comment]
Hi @WeaxMe 

`any()` operator is quite tricky, so we always discourage to do use it.
In this case, the best solution is to just use dot notation, eg.

* if you know the field name
```
SELECT FROM TestEmbedded WHERE embedded.name = 'test_2'
```
* if you don't know the field name and you are interested in any fields that have that value:
```
SELECT FROM TestEmbedded WHERE embedded.values() CONTAINS 'test_2'
```

I hope it helps

Thanks

Luigi

**Comment by WeaxMe** [comment]
Hi @luigidellaquila 
In most cases I don't know the field name and I can  not use SQL like this:

`SELECT FROM TestEmbedded WHERE embedded.name = 'test_2'`

I test SQL:

`SELECT FROM TestEmbedded WHERE embedded.values() CONTAINS 'test_2'`

and it's don't work. I get empty result.
If I execute SQL:

`SELECT embedded.values() FROM TestEmbedded'`

I get empty result too.

Thanks,
Vitaliy.

**Comment by luigidellaquila** [comment]
Ok, this could be a bug, let me check it...

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Oh, ok, I got the problem, `embedded` is an embedded document, not an embedded map.
I think I can make `values()` also manage document field values. I'll update you soon

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @WeaxMe 

I just pushed a fix on 2.2.x and develop branches. It will be released with 2.2.30

Thanks

Luigi

**Comment by WeaxMe** [comment]
Thank you
